src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -101,8 +104,43 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
             using (var telemetry = TelemetryActivity.CreateTelemetryActivityWithNewOperationIdAndEvent(parentId: ParentId, eventName: ProjectRestoreInformation))
             {
                 _operationId = telemetry.OperationId;
+
+                var cpvmEnabled = _request.Project.RestoreMetadata?.CentralPackageVersionsEnabled ?? false;
+                telemetry.TelemetryEvent[CentralVersionManagementEnabled] = cpvmEnabled;
+
                 var restoreTime = Stopwatch.StartNew();
 
+                // The dependencies should not have versions explicitelly defined if cpvm is enabled.
+                if (cpvmEnabled)
+                {
+                    var dependenciesWithDefinedVersion = _request.Project.TargetFrameworks.SelectMany(tfm => tfm.Dependencies.Where(d => !d.LibraryRange.VersionRange.IsCentral && !d.AutoReferenced));
+                    if (dependenciesWithDefinedVersion.Any())
+                    {
+                        await _logger.LogAsync(RestoreLogMessage.CreateError(NuGetLogCode.NU5200, string.Format(CultureInfo.CurrentCulture, Strings.Error_CentralPackageVersions_VersionsNotAllowed, _request.Project.Name, string.Join(";", dependenciesWithDefinedVersion.Select(d => d.Name)))));
```

Comment:
Do we need the project name here? Errors are usually associated with the failing projects without including them in the error message. 
I could be wrong, but check how the NU1605 and equivalents work.

Reply:
> Don't you need to add this log message to the assets file?

It will need to be propagated in VS IDE (adding it to assets or other way). This is related with issue : https://github.com/NuGet/Home/issues/7717 

Currently it will display the error on the command line restore. 



> Do we need the project name here? Errors are usually associated with the failing projects without including them in the error message.
> I could be wrong, but check how the NU1605 and equivalents work.

Removed the project.